rpc/rpcserver/server.go

```diff
@@ -2031,7 +2050,70 @@ func StartTicketBuyerService(server *grpc.Server, loader *loader.Loader, tbCfg *
 	}
 }
 
-func (t *ticketbuyerServer) checkReady() bool {
+// StartTicketBuyer starts the automatic ticket buyer for the v2 service.
+func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr pb.TicketBuyerV2Service_RunTicketBuyerServer) error {
+
+	wallet, ok := t.loader.LoadedWallet()
+	if !ok {
+		return status.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+	params := wallet.ChainParams()
+
+	// Confirm validity of provided voting addresses and pool addresses.
+	var votingAddress dcrutil.Address
+	var err error
+	if req.VotingAddress != "" {
+		votingAddress, err = decodeAddress(req.VotingAddress, params)
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "Invalid Voting Address")
+		}
+	}
+	var poolAddress dcrutil.Address
+	if req.PoolAddress != "" {
+		poolAddress, err = decodeAddress(req.PoolAddress, params)
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "Invalid Pool Address")
+		}
+	}
+
+	if req.BalanceToMaintain < 0 {
+		return status.Errorf(codes.InvalidArgument, "Negative balance to maintain given")
+	}
+
+	tb := tbv2.New(wallet)
+
+	// Set ticketbuyerV2 config
+	tb.AccessConfig(func(c *tbv2.Config) {
+		c.Account = req.Account
+		c.VotingAccount = req.VotingAccount // TODO: Make this a unique config option. Set to acct for compat with v1.
+		c.Maintain = dcrutil.Amount(req.BalanceToMaintain)
+		c.VotingAddr = votingAddress
+		c.PoolFeeAddr = poolAddress
+		c.PoolFees = req.PoolFees
+	})
+
+	// Make sure to lock the wallet on context cancellation.
+	var lockWallet func()
+	lock := make(chan time.Time, 1)
+	lockWallet = func() {
+		lock <- time.Time{}
+		zero.Bytes(req.Passphrase)
+	}
+	tb.SetLock(lock)
```

Comment:
what's the point of all of this, rather than just deferring the lock to happen once Run finishes and this method returns?